framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -1579,10 +1583,19 @@ public ParameterizedExecutableType methodFromUse(MethodInvocationTree tree) {
         if (dependentTypesHelper != null) {
             dependentTypesHelper.viewpointAdaptMethod(tree, method);
         }
-        poly.resolve(tree, method);
+
         return mType;
     }
 
+    @Override
+    public void methodFromUsePreSubstitution(ExpressionTree tree, AnnotatedTypeMirror mirror) {
+        // no call to super needed since super is no-op
+        super.methodFromUsePreSubstitution(tree, mirror);
+        if (tree instanceof MethodInvocationTree && mirror instanceof AnnotatedExecutableType) {
```

Comment:
What other trees/types could be passed in here?
The method name makes it sound like this would only ever make sense for these more specific types.